### PR TITLE
Add loadFeatureFlagsOnStart configuration flag

### PR DESCRIPTION
**Pull request by marioradonic**
**What does this PR do?**

It adds a configuration flag similar to JS library's `advanced_disable_feature_flags_on_first_load` with which a first load of feature flags can be disabled.

**Where should the reviewer start?**

Not much to this PR. Variable name can be discussed and I'm not sure if this is the best way to write the test, but I've tried to follow current structure. I didn't wanna name the variable the same as it's in JS lib since in iOS we don't have a "page load" concept.

**How should this be manually tested?**

I tested it manually by intercepting http requests using Charles proxy and confirming that /decide call is not made.

**Any background context you want to provide?**

In our project we want to be in control of when to reload feature flags. The flag is true by default so it's backward compatible.

**What are the relevant tickets?**
/

**Screenshots or screencasts (if UI/UX change)**
/

**Questions:**
- Does the docs need an update?

Could be useful, but not my job.

- Are there any security concerns?

I don't see any.

- Do we need to update engineering / success?

/

### CODE REVIEW

File: PostHog/Classes/PHGPostHogConfiguration.h [hunk 1]
```diff
@@ -123,6 +123,13 @@ typedef NSMutableURLRequest *_Nonnull (^PHGRequestFactory)(NSURL *_Nonnull);
  */
 @property (nonatomic, assign) BOOL shouldSendDeviceID;
 
+/**
+ * Whether the posthog client should load feature flags when initialized.
+ * Changing the value of this property after initializing the client will have no effect.
+ * The default value is `YES`.
+ */
+@property (nonatomic, assign) BOOL loadFeatureFlagsOnStart;
```
Comment (marandaneto) [inline PostHog/Classes/PHGPostHogConfiguration.h:131]:
```suggestion
@property (nonatomic, assign) BOOL preloadFeatureFlags;
```

Let's call it `preloadFeatureFlags` as the other SDKs.

File: PostHog/Classes/PHGPostHogConfiguration.h [hunk 2]
```diff
@@ -123,6 +123,13 @@ typedef NSMutableURLRequest *_Nonnull (^PHGRequestFactory)(NSURL *_Nonnull);
  */
 @property (nonatomic, assign) BOOL shouldSendDeviceID;
 
+/**
+ * Whether the posthog client should load feature flags when initialized.
+ * Changing the value of this property after initializing the client will have no effect.
```
Comment (marandaneto) [inline PostHog/Classes/PHGPostHogConfiguration.h:128]: We can add a comment that they can be reloaded by calling the `reloadFeatureFlags` method.

File: PostHog/Classes/PHGPostHogConfiguration.h (hunk 1 above)
Comment (marioradonic) [reply PostHog/Classes/PHGPostHogConfiguration.h]: Updated.

File: PostHog/Classes/PHGPostHogConfiguration.h (hunk 2 above)
Comment (marioradonic) [reply PostHog/Classes/PHGPostHogConfiguration.h:128]: Updated.

**Review by marandaneto — APPROVED** [review]

**Pull request merged**